engine/table/src/main/java/io/deephaven/engine/table/impl/util/DynamicTableWriter.java

```diff
@@ -481,7 +481,7 @@ void writeToColumnSource() {
 
         @Override
         public void set(Object value) {
-            setByte(value == null ? QueryConstants.NULL_BYTE : (Byte) value);
+            setByte(value == null ? QueryConstants.NULL_BYTE : ((Number) value).byteValue());
```

Comment:
I kind of hate this set of changes. That said, I also kind of hate that we've documented using the weakly-typed `DynamicTableWriter#logRow` method as if it were advisable; it will never be a high-performance solution to anything.

Maybe a compromise would be to add a new `addPermissive` interface method or similar, that lets the user explicitly send weakly-typed inputs that we'll convert on a best-effort basis.

Reply:
The more I think about it, the more I think adding a weakly-typed parallel code path for Python is the way to go. That way we have Java code that owns the translation from Python.